### PR TITLE
data-ingest endpoint points to local k8s AG service (instead of ClusterAG) if AG data-ingest capability is enabled

### PR DESCRIPTION
**Pull request by aorcholski**

### CODE REVIEW

File: controllers/ingestendpoint/secret.go [hunk 1]
```diff
@@ -111,8 +113,48 @@ func (g *EndpointSecretGenerator) PrepareFields(ctx context.Context, dk *dynatra
...
+		return "", err
+	}
+	return fmt.Sprintf("https://%s-activegate.%s/e/%s/api/v2/metrics/ingest", dk.Name, dk.Namespace, tenant), nil
+}
+
+func extractTenant(url *url.URL) (string, error) {
+	tenant := ""
+	subdomains := strings.Split(url.Hostname(), ".")
+	subpaths := strings.Split(url.Path, "/")
+	//Path = '/e/<token>/api' -> ['', 'e',  '<token>', 'api']
```
Comment (mjgrzybek) [inline controllers/ingestendpoint/secret.go:151]: `token` -> `tenant`

File: controllers/ingestendpoint/secret.go [hunk 2]
```diff
@@ -111,8 +113,48 @@ func (g *EndpointSecretGenerator) PrepareFields(ctx context.Context, dk *dynatra
...
+	url, err := url.Parse(dk.Spec.APIURL)
+	if err != nil {
+		return "", errors.WithMessage(err, "failed to parse DynaKube.spec.apiUrl")
+	}
+
+	tenant, err := extractTenant(url)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("https://%s-activegate.%s/e/%s/api/v2/metrics/ingest", dk.Name, dk.Namespace, tenant), nil
```
Comment (mjgrzybek) [inline controllers/ingestendpoint/secret.go:144]: replace hardcoded `"activegate"` with `dynatracev1beta1.DataIngestCapability.ShortName`

**Review by mjgrzybek — CHANGES_REQUESTED** [review]

File: controllers/ingestendpoint/secret.go [hunk 3]
```diff
@@ -111,8 +113,48 @@ func (g *EndpointSecretGenerator) PrepareFields(ctx context.Context, dk *dynatra
 		dataIngestToken = string(token)
 	}
 
+	diUrl := ""
+	var err error
+	if dk.ActiveGateCapabilityMode(dynatracev1beta1.DataIngestCapability.ShortName) {
+		if diUrl, err = diUrlFromApiUrl(dk); err != nil {
```
[the remark this answers is not in the document]
Comment (aorcholski) [reply controllers/ingestendpoint/secret.go:119]: renamed

File: api/v1beta1/properties.go [hunk 4]
```diff
@@ -68,6 +68,15 @@ func (dk *DynaKube) ActiveGateMode() bool {
 	return len(dk.Spec.ActiveGate.Capabilities) > 0
 }
 
+func (dk *DynaKube) ActiveGateCapabilityMode(capability string) bool {
+	for _, cap := range dk.Spec.ActiveGate.Capabilities {
```
[the remark this answers is not in the document]
Comment (aorcholski) [reply api/v1beta1/properties.go:72]: renamed

File: controllers/ingestendpoint/secret.go (hunk 2 above)
Comment (aorcholski) [reply controllers/ingestendpoint/secret.go]: replaced

**Review by mjgrzybek — DISMISSED** [review]

**Review by mjgrzybek — APPROVED** [review]

File: controllers/ingestendpoint/secret.go [hunk 5]
```diff
@@ -111,8 +115,53 @@ func (g *EndpointSecretGenerator) PrepareFields(ctx context.Context, dk *dynatra
...
+	}
+
 	return map[string]string{
-		UrlSecretField:   fmt.Sprintf("%s/v2/metrics/ingest", dk.Spec.APIURL),
+		UrlSecretField:   diUrl,
 		TokenSecretField: dataIngestToken,
 	}, nil
 }
+
+func getDataIngestUrlFromDk(dk *dynatracev1beta1.DynaKube) (string, error) {
```
Comment (0sewa0) [inline controllers/ingestendpoint/secret.go:129]: small nitpick:
rename to `getDataIngestUrlFromDynakube` so its clearer

File: controllers/ingestendpoint/secret.go [hunk 6]
```diff
@@ -111,8 +115,53 @@ func (g *EndpointSecretGenerator) PrepareFields(ctx context.Context, dk *dynatra
 		dataIngestToken = string(token)
 	}
 
+	diUrl, err := getDataIngestUrlFromDk(dk)
```
Comment (0sewa0) [inline controllers/ingestendpoint/secret.go:118]: small nitpick:
rename to `dataIngestUrl` so its clearer (also its more consistent with other names such as `dataIngestToken`)

**Review by 0sewa0 — CHANGES_REQUESTED** [review]

File: controllers/ingestendpoint/secret.go (hunk 5 above)
Comment (aorcholski) [reply controllers/ingestendpoint/secret.go]: renamed

File: controllers/ingestendpoint/secret.go (hunk 6 above)
Comment (aorcholski) [reply controllers/ingestendpoint/secret.go]: Functions renamed and small wrapper added so I hope it is clearer now

Comment (0sewa0) [reply controllers/ingestendpoint/secret.go]: actually I meant you to just rename `diUrl` to `dataIngestUrl`, but this is nicer :joy:

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**